edx/edx-enterprise
# ENT-362 Fixing inactivation of courses that are removed from catalogs

**Description:** We weren't sending all of the required fields for a course update, which causes the entire call to fail even for properly formatted courses in the same payload.

**JIRA:** https://openedx.atlassian.net/browse/ENT-362

**Dependencies:** n/a

**Merge deadline:** List merge deadline (if any)

**Installation instructions:** Requires an environment with the catalog service running.

**Testing instructions:**

Edxapp sandbox: https://brittneyexline.sandbox.edx.org
Discovery sandbox: https://discovery-brittneyexline.sandbox.edx.org

1. Set up a configuration for an enterprise customer and success factors against our test SAP instance. (This has been done already on the sandbox)
2. Make a catalog with two or more courses. (On the sandbox it is "One Catalog to rule them all")
3. Run the catalog export script, ensure that it has run successfully. To do this on the sandbox, ssh to the edxapp sandbox. Then run the following commands:

`$ sudo -H -u edxapp bash`
`$ source /edx/app/edxapp/edxapp_env`
`$ cd /edx/app/edxapp/edx-platform/`
`$ /edx/bin/python.edxapp /edx/bin/manage.edxapp lms transmit_courseware_data --catalog_user staff --settings=aws`

To see the courses on SAP, follow the instructions here for Importing Courses to view which courses have been sent.
https://openedx.atlassian.net/wiki/pages/viewpage.action?spaceKey=SOL&title=SAP+Testing+and+Configuration+notes

4. Remove one of the courses from the catalog.
5. Run the script again, and observe that the call to SAP was successful, and that the removed course was sent as INACTIVE. Check SAP's interface, the course that was removed should not show up in the Imported Courses search view.

**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.

**Post merge:**

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.
